calliope/core/preprocess/model_data.py

```diff
@@ -127,7 +127,7 @@ def _get_set(constraint):
             return 'loc_techs_supply_plus'
         elif 'resource' in constraint: # i.e. everything with 'resource' in the name that isn't resource_cap
             return 'loc_techs_finite_resource'
-        elif 'storage' in constraint or 'charge_rate' in constraint:
+        elif 'storage' in constraint or 'charge_rate' in constraint or 'energy_cap_per_storage_cap_max' in constraint:
```

Comment:
man, these are such hacks... seeing as 'storage' is in 'energy_cap_per_storage_cap_max' (and '_equals'), this will actually catch those with the first condition (`'storage' in constraint`).

Reply:
True... any ideas on how to improve this?

Reply:
For the time being, no... Stefan and I left it as is when putting 0.6 together because we couldn't think of a non-hardcoded way of doing it